src/main/java/com/limechain/network/ConnectionManager.java

```diff
@@ -25,22 +27,42 @@ public void addPeer(PeerId peerId, PeerInfo peerInfo) {
         peers.put(peerId, peerInfo);
     }
 
+    public PeerInfo getPeerInfo(PeerId peerId) {
+       return peers.get(peerId);
+    }
+
     public void updatePeer(PeerId peerId, BlockAnnounceMessage blockAnnounceMessage) {
-        if (!blockAnnounceMessage.isBestBlock()) {
-            return;
+        final var peer = peers.get(peerId);
+        updateLatestBlock(peer, blockAnnounceMessage.getHeader().getBlockNumber());
+
+        if(blockAnnounceMessage.isBestBlock()) {
```

Comment:
Put a space between 'if' and '(', this goes for all such occurrences. Might be best to see if this can be added to the checkstyle as a rule so that it's more easily checked.